src/components/Chat/ChangePathDialog.tsx

```diff
@@ -0,0 +1,323 @@
+import { useId } from '@floating-ui/react';
+import { ChangeEvent, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { useTranslation } from 'next-i18next';
+
+import classNames from 'classnames';
+
+import {
+  notAllowedSymbols,
+  notAllowedSymbolsRegex,
+} from '@/src/utils/app/file';
+import {
+  filterFoldersWithSearchTerm,
+  getChildAndCurrentFoldersIdsById,
+  getPathToFolderById,
+} from '@/src/utils/app/folders';
+import { PublishedWithMeFilter } from '@/src/utils/app/search';
+
+import { SharingType } from '@/src/types/share';
+import { Translation } from '@/src/types/translation';
+
+import {
+  ConversationsActions,
+  ConversationsSelectors,
+} from '@/src/store/conversations/conversations.reducers';
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks';
+import {
+  PromptsActions,
+  PromptsSelectors,
+} from '@/src/store/prompts/prompts.reducers';
+import { UIActions } from '@/src/store/ui/ui.reducers';
+
+import { MAX_CHAT_AND_PROMPT_FOLDERS_DEPTH } from '@/src/constants/folders';
+
+import CaretIconComponent from '@/src/components/Common/CaretIconComponent';
+import { ErrorMessage } from '@/src/components/Common/ErrorMessage';
+import Modal from '@/src/components/Common/Modal';
+import { NoResultsFound } from '@/src/components/Common/NoResultsFound';
+import Folder from '@/src/components/Folder/Folder';
+
+import FolderPlus from '@/public/images/icons/folder-plus.svg';
+
+interface Props {
+  type: SharingType;
+  isOpen: boolean;
+  onClose: (path: string | undefined) => void;
+  initiallySelectedFolderId: string;
+  depth?: number;
+}
+
+export const ChangePathDialog = ({
+  isOpen,
+  onClose,
+  type,
+  initiallySelectedFolderId,
+  depth,
+}: Props) => {
+  const dispatch = useAppDispatch();
+
+  const { t } = useTranslation(Translation.Chat);
+
+  const headingId = useId();
+
+  const [searchQuery, setSearchQuery] = useState('');
+  const [isAllFoldersOpened, setIsAllFoldersOpened] = useState(true);
+  const [openedFoldersIds, setOpenedFoldersIds] = useState<string[]>([]);
+  const [selectedFolderId, setSelectedFolderId] = useState<string | undefined>(
+    '',
+  );
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
+
+  const { selectors, actions } =
+    type === SharingType.Conversation || type === SharingType.ConversationFolder
+      ? { selectors: ConversationsSelectors, actions: ConversationsActions }
+      : { selectors: PromptsSelectors, actions: PromptsActions };
+
+  const newFolderId = useAppSelector(selectors.selectNewAddedFolderId);
+  const folders = useAppSelector((state) =>
+    selectors.selectTemporaryAndFilteredFolders(state, PublishedWithMeFilter),
+  );
+
+  const filteredFolders = useMemo(
+    () => filterFoldersWithSearchTerm(folders, searchQuery),
+    [folders, searchQuery],
+  );
```

Comment:
`selectTemporaryAndFilteredFolders` already has `searchTerm` parameter can we execute `filterFoldersWithSearchTerm` inside `selectTemporaryAndFilteredFolders` selector?